### PR TITLE
010 person

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/models/orgs.py [hunk 1]
```diff
@@ -87,6 +96,25 @@ def getModelDefs(self):
...
+                        'doc': '',
+                    }),
+                    ('person', ('ps:person', {}), {
+                        'ro': True,
+                        'doc': '',
+                    }),
+                    ('title', ('str', {'lower': True, 'strip': True}), {
+                        'doc': ''
+                    }),
+                    ('start', ('time', {}), {
```
Comment (vertexmc) [inline synapse/models/orgs.py]: start/end are time:min/max

Comment (vEpiphyte) [reply synapse/models/orgs.py]: ACK

File: synapse/models/person.py [hunk 2]
```diff
@@ -1,41 +1,238 @@
...
+                }),
+                ('ps:person:has', ('comp', {'fields': (('person', 'ps:person'), ('node', 'ndef'))}), {
+                    'doc': 'A person owns, controls, or has exclusive use of an object or'
+                           ' resource, potentially during a specific period of time.'
+                }),
+                ('ps:persona:has', ('comp', {'fields': (('persona', 'ps:persona'), ('node', 'ndef'))}), {
+                    'doc': 'A persona owns, controls, or has exclusive use of an object or'
+                           ' resource, potentially during a specific period of time.'
+                }),
+                # FIXME requires file:bytes
```
Comment (vertexmc) [inline synapse/models/person.py]: file:bytes is in 010 now

File: synapse/models/person.py [hunk 3]
```diff
@@ -1,41 +1,238 @@
...
+                    }),
+                    ('node', ('ndef', {}), {
+                        'ro': True,
+                        'doc': 'The object or resource that is owned or controlled by the person.',
+                    }),
+                    ('node:form', ('str', {}), {
+                        'ro': True,
+                        'doc': 'The form of the object or resource that is owned or controlled by the person.',
+                    }),
+                    # FIXME Add seen:min and seen:max
```
Comment (vertexmc) [inline synapse/models/person.py]: Are we going to keep seen:min/max? I thought these were going to become universal props

File: synapse/models/person.py [hunk 4]
```diff
@@ -1,41 +1,238 @@
...
+                    # ('seen:min', {'ptype': 'time:min',
+                    #               'doc': 'The earliest known time when the person owned or controlled the resource.'}),
+                    # ('seen:max', {'ptype': 'time:max',
+                    #               'doc': 'The most recent known time when the person owned or controlled the resource.'}),
+                )),
+                ('ps:persona:has', {}, (
+                    ('persona', ('ps:person', {}), {
```
Comment (vertexmc) [inline synapse/models/person.py]: type should be ps:persona

File: synapse/models/person.py [hunk 5]
```diff
@@ -1,41 +1,238 @@
...
+                        'doc': 'The person name listed for the contact',
+                    }),
+                    ('title', ('str', {'lower': True, 'strip': True}), {
+                        'doc': 'The job/org title listed for this contact.',
+                    }),
+                    # FIXME requires file:bytes
+                    # ('photo', ('file:bytes', {}), {
+                    #     'doc': 'The photo listed for this contact.',
+                    # }),
+                    ('orgname', ('ou:name', {}), {
```
Comment (vertexmc) [inline synapse/models/person.py]: This was a str:lwr before, but probably should have been ou:name

File: synapse/tests/test_lib_chop.py [hunk 6]
```diff
@@ -36,3 +35,22 @@ def test_chop_hexstr(self):
                 self.eq(s_chop.hexstr(v), e)
             else:
                 self.raises(e, s_chop.hexstr, v)
+
+    def test_chop_onespace(self):
+        print('')
```
Comment (vertexmc) [inline synapse/tests/test_lib_chop.py]: would remove print

Comment (vEpiphyte) [reply synapse/tests/test_lib_chop.py]: woops

File: synapse/models/person.py (hunk 5 above)
Comment (vEpiphyte) [reply synapse/models/person.py]: I felt it made sense to use ou:name here

File: synapse/models/person.py (hunk 4 above)
Comment (vEpiphyte) [reply synapse/models/person.py]: ack

File: synapse/models/person.py (hunk 3 above)
Comment (vEpiphyte) [reply synapse/models/person.py]: i think we can drop all these comments when that lands.

File: synapse/models/orgs.py (hunk 1 above)
Comment (vEpiphyte) [reply synapse/models/orgs.py]: fixed

File: synapse/models/person.py (hunk 4 above)
Comment (vEpiphyte) [reply synapse/models/person.py]: fixed

File: synapse/models/person.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/models/person.py]: i want to add file:bytes when we've stablized on what file:bytes is going to be

File: synapse/models/person.py [hunk 7]
```diff
@@ -1,41 +1,238 @@
...
+                ('ps:name', {}, ()),
+                ('ps:person', {}, (
+                    # FIXME aliases?
+                    # ('guidname', ('str', {'lower': True}), {
+                    #     'doc': 'The GUID resolver alias for this person',
+                    # }),
+                    ('dob', ('time', {}), {
+                        'doc': 'The Date of Birth (DOB) if known.',
+                    }),
+                    # FIXME need file:bytes
```
[the remark this answers is not in the document]
Comment (vEpiphyte) [reply synapse/models/person.py]: i want to add file:bytes when we've stablized on what file:bytes is going to be

File: synapse/models/person.py (hunk 5 above)
Comment (vertexmc) [reply synapse/models/person.py]: agreed

File: synapse/models/orgs.py [hunk 8]
```diff
@@ -99,20 +99,20 @@ def getModelDefs(self):
...
-                        'doc': '',
+                        'doc': 'The GUID of the person that is a member of an org.',
                     }),
                     ('title', ('str', {'lower': True, 'strip': True}), {
-                        'doc': ''
+                        'doc': 'The persons normalized title.'
                     }),
-                    ('start', ('time', {}), {
-                        'doc': '',
+                    ('start', ('time:min', {}), {
```
Comment (vertexmc) [inline synapse/models/orgs.py]: Times should be like: ('time', {'min': True})

**Pull request merged**